src/components/manage/Blocks/Listing/ListingBody.jsx

```diff
@@ -17,14 +17,15 @@ const ListingBody = ({ data, properties, intl, path, isEditMode }) => {
     (state) => state.querystringsearch.subrequests,
   );
   const dispatch = useDispatch();
+  const queryLength = data?.query?.length || 0;
 
   React.useEffect(() => {
-    if (data?.query?.length > 0) {
+    if (queryLength > 0) {
       dispatch(
         getQueryStringResults(path, { ...data, fullobjects: 1 }, data.block),
       );
     }
-  }, [dispatch, data, data.block, path]);
+  }, [data, dispatch, path, queryLength]);
```

Comment:
`queryLenght` will be different on every rerender. We don't want the action happen every time that happen.

Reply:
``data`` was already a dependency to the useEffect. I thought I might be able to eliminate the dependency on data with a ``useCallback``, but of course, once ``data`` changes for useCallback, the callback function changes and the effect is called again.

Reply:
@tiberiuichim yeah, I've been in the useCallback/useMemo story! No luck :( 
`data.block` it used to work well (no render when selecting the block).
I have to admit that adding Redux and `useSelector` in play, the scenario is more difficult and forces to write `useEffect` in an specific way.